src/frontends/pytorch/src/op/div.cpp

```diff
@@ -14,21 +19,27 @@ namespace op {
 OutputVector translate_div(NodeContext& context) {
     auto x = context.get_input(0);
     auto y = context.get_input(1);
-    auto res = context.mark_node(std::make_shared<opset10::Divide>(x, y, true));
+    std::string rounding_mode = "";
     if (!context.input_is_none(2)) {
-        auto rounding_mode = context.const_input<std::string>(2);
-        if (rounding_mode == "floor") {
-            res = context.mark_node(std::make_shared<opset10::Floor>(res));
-        } else if (rounding_mode == "trunc") {
-            const auto convert = context.mark_node(std::make_shared<opset10::Convert>(res, element::i64));
-            res = context.mark_node(std::make_shared<opset10::ConvertLike>(convert, x));
-        } else {
-            FRONT_END_OP_CONVERSION_CHECK(false,
-                                          "Openvino Pytorch Frontend doesn't support rounding mode ",
-                                          rounding_mode,
-                                          " for aten::div");
+        rounding_mode = context.const_input<std::string>(2);
+    }
+    if (rounding_mode.empty()) {
+        // if no rounding mode and both inputs are ints cast BOTH to fp32
+        const auto x_dtype = x.get_element_type();
+        const auto y_dtype = y.get_element_type();
+        if (x_dtype.is_static() && x_dtype.is_integral() && y_dtype.is_static() && y_dtype.is_integral()) {
+            x = context.mark_node(std::make_shared<v0::Convert>(x, element::f32));
+            y = context.mark_node(std::make_shared<v0::Convert>(y, element::f32));
```

Comment:
may be it is better always cast them into `f32` without those checks? What if the type is dynamic and during inference they will be of `int32` type?

Reply:
No, fp64 should be left as fp64, also mixed case with int x float is better handled by `align_eltwise_input_types`